rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/service/AzureImportAuditsClient.java

```diff
@@ -71,19 +74,37 @@ public AzureImportAuditsClient(CloudBlobContainer cloudBlobContainer, Applicatio
         String currentDateTime;
         Integer azureImportAuditsGetLimit = applicationParams.getAzureImportAuditsGetLimit();
 
-        while (audits.size() < azureImportAuditsGetLimit) {
+        int counter = azureImportAuditsGetLimit;
+
+        // Check 10 days at least + 5 days for each audit to collect.
+        int maxDaysToCheck = 10 + azureImportAuditsGetLimit * 5;
+        int daysChecked = 0;
+
+        while ((audits.size() < azureImportAuditsGetLimit) && (daysChecked <= maxDaysToCheck)) {
+
             currentDateTime = localDateTime.format(DateTimeFormatter.ofPattern(DATE_PATTERN));
+
             ResultSegment<ListBlobItem> blobsPage = cloudBlobContainer.listBlobsSegmented(currentDateTime,
                 FLAT_BLOB_LISTING,
                 ONLY_COMMITTED_BLOBS,
                 Integer.MAX_VALUE,
                 NO_CONTINUATION_TOKEN,
                 NO_OPTIONS,
                 NO_OP_CONTEXT);
+
             localDateTime = localDateTime.minus(1, ChronoUnit.DAYS);
-            List<ImportAudit> auditsLastBatch = populateListOfAudits(blobsPage);
 
-            audits.addAll(auditsLastBatch);
+            if (blobsPage != null && !blobsPage.getResults().isEmpty()) {
+                List<ImportAudit> auditsLastBatch = populateListOfAudits(blobsPage);
+
+                audits.addAll(auditsLastBatch);
+            }
+
+            if (audits.isEmpty() && (daysChecked == maxDaysToCheck)) {
+                log.error("No import audits found over {} iterations", maxDaysToCheck);
+            }
```

Comment:
This conditional log inside the loop just hinders smoother readability.
Could you please take it outside, after the loop?

Reply:
Completed. Error log moved to outside of the while loop

Reply:
Removed unused variable as flagged by Sonar scan